### PR TITLE
Add namespace to datadog helm chart templates

### LINKED ISSUE #217: Helm template with non default namespace generates incorrectly
**Describe what happened:**
When making an K8S file with `helm template --namespace datadog datadog-agent datadog/datadog`, the resulting files are incorrect.
Some resources do have an namespace set, some don't. For example, the cluster-agent-rbac has an subject to the ServiceAccount in the namespace "datadog", but mentioned ServiceAccount doesn't have an explicit namespace and thus will be created in the current active namespace.

**Describe what you expected:**
I expected the resources which are scoped to a namespace, be defined in given namespace.

**Steps to reproduce the issue:**
- Run `helm template --namespace datadog datadog-agent datadog/datadog`
- Observe the lack of namespace on many resources (ServiceAccounts, Services, Deployments, RoleBindings, Daemonsets, ConfigMaps)

**Additional environment details (Operating System, Cloud provider, etc):**
None

### PR DESCRIPTION
**Pull request by afirth**
#### What this PR does / why we need it:
Namespaces are unsettable on ~1/2 the resources

#### Which issue this PR fixes

  - fixes #217 

#### Special notes for your reviewer:

I'm not 100% sure I got them all - you might want to add some tests.

#### Checklist

- [x] Chart Version bumped
- [x] `CHANGELOG.md` has beed updated

### CODE REVIEW

File: charts/datadog/templates/rbac.yaml [hunk 1]
```diff
@@ -2,13 +2,14 @@
 apiVersion: {{ template "rbac.apiVersion" . }}
 kind: ClusterRole
 metadata:
+  name: {{ template "datadog.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (clamoriniere) [inline charts/datadog/templates/rbac.yaml:6]: I think ClusterRole doesn't need a namespace

**Review by clamoriniere — COMMENTED** [review]
Thank you for opening this PR. 
Looks almost perfect (only one comment about clusterRole)

File: charts/datadog/templates/rbac.yaml (hunk 1 above)
Comment (afirth) [reply charts/datadog/templates/rbac.yaml]: :100: fixed in 959a9f5

**Comment by clamoriniere** [comment]
I looked into every resources present in the chart. the namespace is missing in few places:
- kube-state-metrics-network-policy.yaml
- kube-state-metrics-cilium-network-policy.yaml
- agent-clusterchecks-cilium-network-policy.yaml

If you can update your PR to add them, it will be awesome.

File: charts/datadog/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +15,7 @@
 * Improve support for environment autodiscovery by removing explicit setting of `DOCKER_HOST` by default with Agent 7.27+.
 Starting Agent 7.27, the recommended setup is to never set `datadog.dockerSocketPath` or `datadog.criSocketPath`, except if your setup is using non-standard paths.
 
+
```
Comment (afirth) [inline charts/datadog/CHANGELOG.md:18]:
```suggestion
```

**Comment by afirth** [comment]
@clamoriniere cheers, dunno why i thought they were cluster scoped. fixed and rebased

**Comment by clamoriniere** [comment]
> @clamoriniere This is done I think, I rebased it once but can't keep up with the version updates. Feel free to push on top of it.

**Comment by clamoriniere** [comment]
sorry for the close and reopen...

**Review by clamoriniere — APPROVED** [review]
Hi @afirth 
thanks for the all the work you made on this PR.

**Pull request merged**